x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/Ccr.java

```diff
@@ -373,10 +373,6 @@ public Optional<EngineFactory> getEngineFactory(final IndexSettings indexSetting
 
     @SuppressWarnings("HiddenField")
     public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
-        if (enabled == false) {
```

Comment:
Not that it matters, but was this change necessary or did you just find the "optimization" to not have the extra thread pool unnecessary?

Reply:
We apparently do some CCR-related things even if CCR is disabled, or at least there are tests which now fail if the CCR threadpool is contingent on whether it's enabled or not. Seemed simplest to just create the threadpool either way rather than pick all that apart.